File: src/stack/lc_150_evaluate_reverse_polish_notation/evaluate_reverse_polish_notation.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        stack = []
        operators = {'+', '-', '*', '/'}
        for token in tokens:
            if token in operators:
                b = stack.pop()
                a = stack.pop()
                match token:
                    case '+':
                        c = a + b
                    case '-':
                        c = a - b
                    case '*':
                        c = a * b
                    case '/':    
                        c = int(a / b)
                stack.append(c)       
            else:
                stack.append(int(token))
                
        return stack[-1]

File: src/stack/lc_150_evaluate_reverse_polish_notation/evaluate_reverse_polish_notation.java
import java.util.Stack;

class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int v1;
        int v2;
        for(int i = 0; i < tokens.length; i++) {
            switch(tokens[i]) {
                case "+":
                    v2 = stack.pop();
                    v1 = stack.pop();
                    stack.push(v1 + v2);
                    break;
                case "-":
                    v2 = stack.pop();
                    v1 = stack.pop();
                    stack.push(v1- v2);
                    break;
                case "*":
                    v2 = stack.pop();
                    v1 = stack.pop();
                    stack.push(v1 * v2);
                    break;
                case "/":
                    v2 = stack.pop();
                    v1 = stack.pop();
                    stack.push(v1 / v2);
                    break;
                default:
                    stack.push(Integer.parseInt(tokens[i]));                
            }
        }
        return stack.peek();
    }
}